### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by Picayune1**
This is a Siamese model designed by Nathan Levu student id 46823748.

The algorithm work's to a certain extent, with 5 epochs getting a 60 % accuracy, however running just 5 epochs took 3 hours even with reduced reducing size of datafile. 

Working algorithm can be run by running the predict.py file.

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: OK
- Algorithm and training choices explained well
- only one plot provided for training set, no axes or legends,  not very useful

Commit messages: Good

Code: 
- it is a bit strange that 5 epochs take as long as 3 hours to train! Must be an issue with the code
- upon closer inspection, it is actually a coding issue - the while loops in Siamese_dataset in dataset.py will keep on running until it hits the if statements to return the last picked img1 (line 31 and 36)
- code design is ok, image pair selection could be more efficient
- minimal comments
- **only implements a siamese network, no classifier**

Functionality/Performance: 
- **only performing similarity detection between pairs of images, no classification**
- siamese netwrk seems to be training poorly, only 60% accuracy, but no actions have been taken to resolve it

General comments:
- **algorithm solves the respective problem only half-way**
- you need to have added a classifier that can classify an individual image as AD or NC with their features extracted by the trained siamese network

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation, bug makes it run slow -1 still achieves results +1
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem partially no classifier -3
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, plot missing details -1
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good but missing refs -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description minimal -1

**Comment by shakes76** [comment]
Merging can only be done if the folder is renamed with more specific name than 'Siamese'. Does not affect grade only merge.

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks lost.
